.github/workflows/run-pytest.yml

```
@@ -16,41 +16,55 @@ concurrency:
   cancel-in-progress: true
 jobs:
   test:
+    name: test (${{ matrix.os }} ${{ matrix.python }} ${{ matrix.dependencies }})
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        config: [ {python: '3.8', dependencies: 'newest'},
-                  {python: '3.9', dependencies: 'newest'},
-                  {python: '3.10', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'minimal'},
-                  {python: '3.8', dependencies: 'oldest'} ]
+        os: [ubuntu-latest]
+        python: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        include:
+          # Default to newest dependencies
+          - dependencies: 'newest'
+          # Test the oldest Python with the oldest supported dependencies.
+          - python: '3.8'
+            dependencies: 'oldest'
+          # Test the newest Python with both the newest and minimal dependencies.
+          - python: '3.12'
+            dependencies: 'newest'
+          - python: '3.12'
+            dependencies: 'minimal'
+          # Test macOS and windows only on latest version
+          - os: 'macos-latest'
+            python: '3.12'
+            dependencies: 'newest'
+          - os: 'windows-latest'
+            python: '3.12'
+            dependencies: 'newest'
```

I'd vote for testing also the oldest versions we claim to support to detect when we should update our minimal supported version.

```suggestion
            dependencies: 'newest'
          - os: 'windows-latest'
            python: 3.8
            dependencies: 'oldest'
          - os: 'macos-latest'
            python: 3.8
            dependencies: 'oldest'
```